cellarium/ml/core/saving.py

```diff
@@ -0,0 +1,53 @@
+# Copyright Contributors to the Cellarium project.
+# SPDX-License-Identifier: BSD-3-Clause
+
+from typing import Any
+
+import lightning.pytorch as pl
+from jsonargparse import ArgumentParser
+from lightning.pytorch.utilities.rank_zero import rank_zero_warn
+
+
+def _load_state(
+    cls: type[pl.LightningModule],
+    checkpoint: dict[str, Any],
+    strict: bool | None = None,
+    **cls_kwargs_new: Any,
+) -> pl.LightningModule:
+    """
+    Re-implementation of :func:`lightning.pytorch.core.saving._load_state` that instantiates the model
+    using the configuration saved in the checkpoint.
+    """
+    ### cellarium.ml - this part is different from the original
+    parser = ArgumentParser()
+    parser.add_class_arguments(cls, "model")
+
+    # pass in the values we saved automatically
+    if cls.CHECKPOINT_HYPER_PARAMS_KEY in checkpoint:
+        cfg = checkpoint[cls.CHECKPOINT_HYPER_PARAMS_KEY]
+    else:
+        cfg = parser.get_defaults()
```

Comment:
Is this worth logging a warning?

Reply:
Ok, I removed reading defaults to be consistent with an upstream implementation and avoid confusion.